pkg/asset/manifests/gcp/cluster.go

```diff
@@ -0,0 +1,140 @@
+		// TODO: common function - this will be used in machine manifest creation too
+		masterSubnet = fmt.Sprintf("%s-%s-subnet", clusterID.InfraID, "master")
+		masterSubnetDescription = "Control Plane Subnet owned by the installer"
+	}
+	master := capg.SubnetSpec{
+		Name:           masterSubnet,
+		CidrBlock:      "",
+		Description:    &masterSubnetDescription,
+		Region:         installConfig.Config.GCP.Region,
+		EnableFlowLogs: &enableFlowLogs,
+		Purpose:        &masterSubnetDescription,
+	}
+
+	workerSubnet := installConfig.Config.GCP.ComputeSubnet
+	workerSubnetDescription := "Compute Subnet used by the installer"
+	if workerSubnet == "" {
+		// TODO: common function - this will be used in machine manifest creation too
+		workerSubnet = fmt.Sprintf("%s-%s-subnet", clusterID.InfraID, "worker")
+		workerSubnetDescription = "Compute Subnet owned by the installer"
+	}
+	worker := capg.SubnetSpec{
+		Name:           workerSubnet,
+		CidrBlock:      "",
+		Description:    &workerSubnetDescription,
+		Region:         installConfig.Config.GCP.Region,
+		EnableFlowLogs: &enableFlowLogs,
+		Purpose:        &workerSubnetDescription,
+	}
+
+	// Add the CIDR information
+	machineV4CIDRs := []string{}
+	for _, network := range installConfig.Config.Networking.MachineNetwork {
+		if network.CIDR.IPNet.IP.To4() != nil {
+			machineV4CIDRs = append(machineV4CIDRs, network.CIDR.IPNet.String())
+		}
+	}
+
+	if len(machineV4CIDRs) == 0 {
+		return nil, fmt.Errorf("failed to parse machine CIDRs")
+	}
+
+	_, ipv4Net, _ := net.ParseCIDR(machineV4CIDRs[0])
+	if installConfig.Config.GCP.ControlPlaneSubnet == "" {
+		masterCIDR, err := cidr.Subnet(ipv4Net, 1, 0)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create the master subnet %s", err.Error())
+		}
+		// The subnet must be created, provide the CIDR
+		master.CidrBlock = masterCIDR.String()
+	}
+
+	if installConfig.Config.GCP.ComputeSubnet == "" {
+		workerCIDR, err := cidr.Subnet(ipv4Net, 1, 1)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create the worker subnet %s", err.Error())
+		}
+		// The subnet must be created, provide the CIDR
+		worker.CidrBlock = workerCIDR.String()
+	}
+
+	subnets := []capg.SubnetSpec{master, worker}
+
+	labels := make(map[string]string, len(installConfig.Config.GCP.UserLabels)+1)
+	// add OCP default label
+	labels[fmt.Sprintf("kubernetes-io-cluster-%s", clusterID.InfraID)] = "owned"
+	for _, label := range installConfig.Config.GCP.UserLabels {
+		labels[label.Key] = label.Value
+	}
+
+	gcpCluster := &capg.GCPCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterID.InfraID,
+			Namespace: capiutils.Namespace,
+		},
+		Spec: capg.GCPClusterSpec{
+			Project: installConfig.Config.GCP.ProjectID,
+			Region:  installConfig.Config.GCP.Region,
+			Network: capg.NetworkSpec{
+				Name:                  &networkName,
+				AutoCreateSubnetworks: &autoCreateSubnets,
+				Subnets:               subnets,
+			},
+			AdditionalLabels: labels,
```

Comment:
Does CredentialsRef need to be set here?
// CredentialsRef is a reference to a Secret that contains the credentials to use for provisioning this cluster. 
// If not supplied then the credentials of the controller will be used.

Reply:
@patrickdillon said that this does not need to be set yet. We can figure this out later (if it does need to be set at all).